src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,176 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.InvalidUserDataException;
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "defaultLogin";
+    private static final String DEFAULT_PASSWORD = "defaultPassword";
+    private static final Integer DEFAULT_AGE = 36;
+    private static final String LESS_THAN_6_CHARACTERS = "short";
+    private static final String EMPTY_STRING = "";
+    private static final String NULL_STRING = null;
+    private static final Integer NULL_INTEGER = null;
+    private static final Integer ZERO_AGE = 0;
+    private static final Integer UNDERAGE = 17;
+    private static final Integer AGE_EIGHTEEN = 18;
+    private static final Integer NEGATIVE_AGE = -1;
+    private static final String SIX_CHARACTER_STRING = "six123";
+    private RegistrationServiceImpl registrationService;
+    private StorageDao storageDao;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        user = new User();
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(DEFAULT_AGE);
+    }
+
+    @ AfterEach
+    void tearDown() {
+        storageDao.clear();
+    }
+
+    @Test
+    void register_validLogin_ok() {
+        user.setLogin(DEFAULT_LOGIN);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_validPassword_ok() {
+        user.setPassword(DEFAULT_PASSWORD);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_validAge_ok() {
+        user.setAge(DEFAULT_AGE);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
```

Comment:
looks like redundant
You can test when user with all valid fields in one test method